mods/default/nodes.lua

```diff
@@ -586,29 +586,121 @@ minetest.register_node("default:snow", {
 	sounds = default.node_sound_snow_defaults(),
+	node_placement_prediction = nil,
 
 	on_construct = function(pos)
-		pos.y = pos.y - 1
-		if minetest.get_node(pos).name == "default:dirt_with_grass" then
-			minetest.set_node(pos, {name = "default:dirt_with_snow"})
+		pos.y = pos.y-1
+		local node = minetest.get_node(pos)
+		if node.name == "default:dirt"
+		or node.name == "default:dirt_with_grass"
+		or node.name == "default:dirt_with_dry_grass"
+		or node.name == "default:dirt_with_grass_footsteps"
+		or node.name == "default:dirt_with_rainforest_litter"
+		or node.name == "default:dirt_with_coniferous_litter" then
+			node.name = "default:dirt_with_snow"
+			minetest.set_node(pos, node)
+		end
+	end,
+
+	-- Handle node drops due to node level.
+	on_dig = function(pos, node, digger)
+		local level = minetest.get_node_level(pos)
+		minetest.node_dig(pos, node, digger)
+
+		if creative and creative.is_enabled_for and creative.is_enabled_for(digger and digger:get_player_name() or "") or
+				minetest.get_node(pos).name == node.name then
+			return
 		end
+
+		if minetest.get_node(pos).name ~= node.name then
+			local inv = digger:get_inventory()
+			if not inv then
+				return
+			end
+			local left = inv:add_item("main", "default:snow "..tostring(level/16-1))
+			if not left:is_empty() then
+				minetest.add_item({
+					x = pos.x + math.random()/2-0.25,
+					y = pos.y + math.random()/2-0.25,
+					z = pos.z + math.random()/2-0.25,
+				}, left)
+			end
+		end
+	end,
+
+	-- Manage snow levels.
+	on_place = function(itemstack, player, pt)
+		local oldnode_under = minetest.get_node_or_nil(pt.under)
+		if not oldnode_under then
+			return itemstack, false
+		end
+
+		-- Dummy def used for olddef_under (unknown nodes treated as solid nodes)
+		local olddef_under = minetest.registered_nodes[oldnode_under.name]
+		olddef_under = olddef_under or {buildable_to = false}
+		if not olddef_under then
+			return itemstack, false
+		end
+
+		-- If node under is buildable_to, place into it instead (eg. snow)
+		local pos, node
+		if olddef_under.buildable_to then
+			pos = pt.under
+			node = oldnode_under
+		else
+			pos = pt.above
+			node = minetest.get_node(pos)
+			local def = minetest.registered_nodes[node.name]
+			if not def
+			or not def.buildable_to then
+				return itemstack, false
+			end
+		end
+
+		if node.name ~= "default:snow" then
+			-- place a snow
+			return minetest.item_place_node(itemstack, player, pt)
+		end
+
+		-- Check if area is protected
+		local playername = player and player:get_player_name() or ""
+		if minetest.is_protected(pos, playername) then
+			minetest.record_protection_violation(pos, playername)
+			return itemstack, false
+		end
+
+		-- make snow thicker
+		local level = minetest.get_node_level(pos)
+		level = level + 16
+		if level < 64 then
+			minetest.set_node_level(pos, level)
+		else
+			-- full cube, replace with snowblock
+			minetest.set_node(pos, {name="default:snowblock"})
+		end
+
+		itemstack:take_item()
```

Comment:
Don't do this if creative is enabled for the placer.

Reply:
Done.

Reply:
Now you only do this if creative is enabled for the player. You have to `not` the condition.

Reply:
I'll change that.